tests/unit/amo/reducers/test_versions.js

```diff
@@ -89,9 +90,26 @@ describe(__filename, () => {
   });
 
   describe('getVersionsBySlug', () => {
-    it('returns null if no versions have been loaded', () => {
+    it('returns an empty array if no versions have been loaded', () => {
       const state = versionsReducer(undefined, { type: 'SOME_OTHER_ACTION' });
-      expect(getVersionsBySlug({ slug: 'some-slug', state })).toBe(null);
+      expect(getVersionsBySlug({ slug: 'some-slug', state })).toEqual([]);
+    });
+
+    it('throws when an expected version is not found for a slug', () => {
+      const slug = 'some-slug';
+      const versions = [fakeVersion];
+      const state = versionsReducer(
+        undefined,
+        loadVersions({ slug, versions }),
+      );
+
+      // We have to manually remove the version in order to trigger the error.
+      delete state.byId[fakeVersion.id];
```

Comment:
I understand this better now. You don't need a test for this. I'll suggest something better than throwing an error.